src/Coordination/KeeperStateMachine.cpp

```diff
@@ -114,6 +113,21 @@ void KeeperStateMachine::init()
         storage = std::make_unique<KeeperStorage>(coordination_settings->dead_session_check_period_ms.totalMilliseconds(), superdigest);
 }
 
+nuraft::ptr<nuraft::buffer> KeeperStateMachine::pre_commit(uint64_t log_idx, nuraft::buffer & data)
+{
+    preprocess(log_idx, data);
+    return nullptr;
+}
+
+void KeeperStateMachine::preprocess(const uint64_t log_idx, nuraft::buffer & data)
```

Comment:
maybe just move this code to `pre_commit` and get rid of this function?

Reply:
Similar comment as in one before, it's going to be changed in #37252 so I hope it's okay to leave it for now.